src/Publisher.php

```diff
@@ -7,14 +7,44 @@
  */
 class Publisher extends Request
 {
+    /**
+    * The result of the call to basic_publish. 
+    * Assumed to be successful unless overridden with 
+    * a basic.return or basic.nack. Only valid when in 
+    * confirm_select mode
+    */
+    private $publish_result = null;
 
     /**
      * @param string  $routing
      * @param Message $message
      * @throws Exception\Configuration
      */
-    public function publish($routing, $message)
+    public function publish($routing, $message, $mandatory=false)
     {
-        $this->getChannel()->basic_publish($message, $this->getProperty('exchange'), $routing);
+	$this->publish_result = true;
+
+	if($mandatory == true) {
+            $this->getChannel()->confirm_select();
+            $this->getChannel()->set_nack_handler([$this, 'nack']);
```

Comment:
If we add the nack handler we should also add the ack handler, what do you think?

Also I would like to make this more configureable.
If I define a ack/nack/return handler it should be set for the channel.